hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -558,18 +559,19 @@ protected InternalScanner createScanner(HStore store, ScanInfo scanInfo,
    * @param cr the compaction request.
    * @param newFiles the new files created by this compaction under a temp dir.
    * @param user the running user.
+   * @param fileAcessor a lambda expression with logic for loading a HStoreFile given a Path.
    * @return A list of the resulting store files already placed in the store dir and loaded into the
    * store cache.
    * @throws IOException if the commit fails.
    */
-  public List<HStoreFile> commitCompaction(CompactionRequestImpl cr, List<Path> newFiles, User user)
-      throws IOException {
+  public List<HStoreFile> commitCompaction(CompactionRequestImpl cr, List<Path> newFiles,
+      User user, Function<Path, HStoreFile> fileAcessor) throws IOException {
```

Comment:
I like the idea of giving the callback -- do you think it would make a nicer API to define our own FunctionalInterface that has some javadoc associated with it? I think it will leave your calling code in HStore as simple, but give the casual observer some more understanding what this function is supposed to do.

Reply:
Definitely, and can make its method throw IOException too, leaving the HStore code cleaner. Had just pushed a new commit with it, for now, created as public inner class of Compactor itself, as it's strictly related to Compactor.